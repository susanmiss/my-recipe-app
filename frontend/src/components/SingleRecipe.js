import React, { useEffect, useState } from 'react'
import axios from 'axios'


const SingleRecipe = (props) => {
    const [recipe, setRecipe, id] = useState('');

    useEffect(() => {
        axios.get(`${process.env.REACT_APP_API}/${props.match.params.id}`)
        .then(response => setRecipe(response.data)) 
        .then(() => console.log('id: ', props.match.params.id))
        .catch(error => alert('Error loading your single deliciuos recipe'))
    }, []);  
    
    const showSingleRecipe = () => (
        <div className="row">
            <div className="col-md-8 offset-md-2 pt-3 pb-2">
                <h1>{recipe.recipeTitle}</h1>
                <p>{recipe.recipeContent}</p>
            </div>
        </div>
    )


    return (
          <div className="container pb-5" data-test="component-singleRecipe">
            {/* {JSON.stringify(props)} */}

            <br />
             {showSingleRecipe()}  
          </div>
          )
}

export default SingleRecipe; 