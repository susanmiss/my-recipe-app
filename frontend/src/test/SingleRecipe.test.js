import React from 'react';
import ReactDOM from 'react-dom'
import {shallow, mount} from 'enzyme'; 
import renderer from 'react-test-renderer';
import SingleRecipe from '../components/SingleRecipe'
import axios from 'axios'


const setup = (props={}) => {
    return shallow(<SingleRecipe {...props} />)
}

const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
}



it('renders correctly', () => {
    const tree = renderer
        .create(<SingleRecipe />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

const wrapper = shallow(<SingleRecipe />)

  
  it('should render a <div />', () => {
      expect(wrapper.find('div').length).toEqual(3);
  });

  it('should render 2 p tags', () =>{
    expect(wrapper.find('p').length).toEqual(1);
  });

  it('should render 1 h1 tag', () =>{
    expect(wrapper.find('h1').length).toEqual(1);
  });
 

