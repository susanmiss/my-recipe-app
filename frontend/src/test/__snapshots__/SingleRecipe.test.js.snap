// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container pb-5"
  data-test="component-singleRecipe"
>
  <br />
  <div
    className="row"
  >
    <div
      className="col-md-8 offset-md-2 pt-3 pb-2"
    >
      <h1 />
      <p />
    </div>
  </div>
</div>
`;
